File: 3 Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.size() < 2)
            return s.size();
        //vector<bool> note(130, 0);//bitset比vector略快
        bitset<130> note;
        
        int cnt = 0;                   //這條件影響不大
        for(int i = 0;i < s.size() && s.size() - i > cnt;++i){
            //note.assign(130, 0);
            note.reset();
            int j;
            for(j = i;j < s.size() && !note[s[j]];++j){
                note[s[j]] = 1;
            }
            if(j - i > cnt)
                cnt = j - i;
        }
        return cnt;
    }
};

File: 3 Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        if len(s) < 2:
            return len(s)

        cnt = 0;
        for i in range(0, len(s)):
            note = [False]*130
            j = i
            while j < len(s) and not(note[ord(s[j])]):
                note[ord(s[j])] = 1
                j += 1
            if j - i > cnt:
                cnt = j - i
        return cnt